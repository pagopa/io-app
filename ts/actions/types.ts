/**
 * Defines Flow types for the available actions and store related stuff.
 */

import { NavigationAction } from "react-navigation";
import {
  Dispatch as DispatchAPI,
  MiddlewareAPI as ReduxMiddlewareAPI,
  Store as ReduxStore,
  StoreEnhancer as ReduxStoreEnhancer
} from "redux";

import { GlobalState } from "../reducers/types";
import { ApplicationActions } from "../store/actions/application";
import { APP_STATE_CHANGE_ACTION } from "../store/actions/constants";
import { ErrorActions } from "../store/actions/error";
import { NotificationsActions } from "../store/actions/notifications";
import { OnboardingActions } from "../store/actions/onboarding";
import { ProfileActions } from "../store/actions/profile";
import { SessionActions } from "../store/actions/session";

export type ApplicationState = "background" | "inactive" | "active";

export type ApplicationStateAction = Readonly<{
  type: typeof APP_STATE_CHANGE_ACTION;
  payload: ApplicationState;
}>;

export type Action =
  | ApplicationActions
  | ApplicationStateAction
  | NavigationAction
  | SessionActions
  | OnboardingActions
  | ProfileActions
  | NotificationsActions
  | ErrorActions;

/* eslint-disable no-use-before-define */
// We need to disable the eslint rule because of a problem described here
// @https://github.com/babel/babel-eslint/issues/485
export type GetState = () => GlobalState;

export type Dispatch = DispatchAPI<Action, GlobalState>;

export type Store = ReduxStore<GlobalState, Action>;

export type StoreEnhancer = ReduxStoreEnhancer<GlobalState>;

export type MiddlewareAPI = ReduxMiddlewareAPI<Dispatch, GlobalState>;

// Props injected by react-redux connect() function
export type ReduxProps = Readonly<{
  dispatch: Dispatch;
}>;
