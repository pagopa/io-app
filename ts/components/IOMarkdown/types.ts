import { VSpacer } from "@pagopa/io-app-design-system";
import {
  TxtHeaderNode,
  TxtListNode,
  TxtListItemNode,
  TxtParagraphNode,
  TxtStrNode,
  ASTNodeTypes,
  TxtStrongNode,
  TxtEmphasisNode,
  TxtLinkNode,
  TxtImageNode,
  TxtBlockQuoteNode,
  TxtCodeBlockNode,
  TxtCodeNode,
  TxtBreakNode,
  TxtHtmlNode,
  TxtHorizontalRuleNode,
  AnyTxtNode
} from "@textlint/ast-node-types";
import { ReactNode, ComponentProps } from "react";

export type Renderer = (txtNode: AnyTxtNodeWithSpacer) => ReactNode;
export type RuleRenderer<P, F extends Renderer | void = void> = (
  param: P,
  renderer: F
) => ReactNode;
export type AnyTxtNodeWithSpacer =
  | AnyTxtNode
  | { type: "Spacer"; size: number; key: string };

type TxtNodeMap = {
  [ASTNodeTypes.Header]: TxtHeaderNode;
  [ASTNodeTypes.Paragraph]: TxtParagraphNode;
  [ASTNodeTypes.Str]: TxtStrNode;
  [ASTNodeTypes.Strong]: TxtStrongNode;
  [ASTNodeTypes.Emphasis]: TxtEmphasisNode;
  [ASTNodeTypes.Link]: TxtLinkNode;
  [ASTNodeTypes.List]: TxtListNode;
  [ASTNodeTypes.ListItem]: TxtListItemNode;
  [ASTNodeTypes.Image]: TxtImageNode;
  [ASTNodeTypes.BlockQuote]: TxtBlockQuoteNode;
  [ASTNodeTypes.CodeBlock]: TxtCodeBlockNode;
  [ASTNodeTypes.Code]: TxtCodeNode;
  [ASTNodeTypes.Break]: TxtBreakNode;
  [ASTNodeTypes.Html]: TxtHtmlNode;
  [ASTNodeTypes.HorizontalRule]: TxtHorizontalRuleNode;
};
type RendererMap = {
  [K in keyof TxtNodeMap]: RuleRenderer<TxtNodeMap[K], Renderer>;
} & {
  [ASTNodeTypes.Comment]: RuleRenderer<null>;
  Spacer: RuleRenderer<ComponentProps<typeof VSpacer> & { key: string }>;
};

export type IOMarkdownRenderRules = RendererMap;
