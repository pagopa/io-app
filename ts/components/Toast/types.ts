import { HapticFeedbackTypes } from "react-native-haptic-feedback";
import { IOIcons } from "@pagopa/io-app-design-system";

export type ToastVariant = "neutral" | "error" | "info" | "success" | "warning";

export type Toast = {
  message: string;
  variant?: ToastVariant;
  icon?: IOIcons;
  hapticFeedback?: keyof typeof HapticFeedbackTypes;
};

export type ToastOptions = Omit<Toast, "message">;
