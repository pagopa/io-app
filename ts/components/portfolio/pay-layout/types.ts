/* subtitlesLS and subtitle are mutually exclusive */
type WithSubtitle = {
  hasMainSubtitle: true;
  hasSubtitlesLR: false;
  subtitleText: string;
};

type WithoutSubtitles = {
  hasMainSubtitle: false;
  hasSubtitlesLR: false;
};

type WithSubtitlesLR = {
  hasSubtitlesLR: true;
  hasMainSubtitle: false;
  subtitleLeftText?: string;
  subtitleRightText?: string;
};

type WithTouchable = {
  hasTouchable: true;
  touchableContent: React.ReactElement<any>;
};

type WithoutTouchable = {
  hasTouchable: false;
};

export type TopContentSubtitle = WithSubtitle & WithoutTouchable;
export type TopContentSubtitlesLR = WithSubtitlesLR & WithoutTouchable;
export type TopContentTouchable = WithoutSubtitles & WithTouchable;

export type TopContentSubtitleTouchable = WithSubtitle & WithTouchable;
export type TopContentSubtitlesLRTouchable = WithSubtitlesLR & WithTouchable;
export type TopContentNone = WithoutSubtitles & WithoutTouchable;

export function topContentSubtitle(subtitleText: string): TopContentSubtitle {
  return {
    hasMainSubtitle: true,
    hasSubtitlesLR: false,
    hasTouchable: false,
    subtitleText
  };
}

export function topContentSubtitlesLR(
  subtitleLeftText?: string,
  subtitleRightText?: string
): TopContentSubtitlesLR {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: true,
    hasTouchable: false,
    subtitleLeftText,
    subtitleRightText
  };
}

export function topContentTouchable(
  touchableContent: React.ReactElement<any>
): TopContentTouchable {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: false,
    hasTouchable: true,
    touchableContent
  };
}

export function topContentSubtitleTouchable(
  touchableContent: React.ReactElement<any>,
  subtitleText: string
): TopContentSubtitleTouchable {
  return {
    hasMainSubtitle: true,
    hasSubtitlesLR: false,
    hasTouchable: true,
    touchableContent,
    subtitleText
  };
}

export function topContentSubtitlesLRTouchable(
  touchableContent: React.ReactElement<any>,
  subtitleLeftText?: string,
  subtitleRightText?: string
): TopContentSubtitlesLRTouchable {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: true,
    hasTouchable: true,
    touchableContent,
    subtitleLeftText,
    subtitleRightText
  };
}

export function topContentNone(): TopContentNone {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: false,
    hasTouchable: false
  };
}

export type TopContent =
  | TopContentSubtitle
  | TopContentSubtitleTouchable
  | TopContentSubtitlesLR
  | TopContentSubtitlesLRTouchable
  | TopContentTouchable
  | TopContentNone;
