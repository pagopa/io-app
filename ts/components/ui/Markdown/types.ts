/**
 * Types of messages sent from inside the webview.
 */

import * as t from "io-ts";

const LinkMessage = t.type({
  type: t.literal("LINK_MESSAGE"),
  payload: t.type({
    href: t.string
  })
});

const ResizeMessage = t.type({
  type: t.literal("RESIZE_MESSAGE"),
  payload: t.type({
    height: t.number
  })
});

const ScrollEndMessage = t.type({
  type: t.literal("SCROLL_END_MESSAGE")
});

export const WebViewMessage = t.taggedUnion("type", [
  LinkMessage,
  ResizeMessage,
  ScrollEndMessage
]);
