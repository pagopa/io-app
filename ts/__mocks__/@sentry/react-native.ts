export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
