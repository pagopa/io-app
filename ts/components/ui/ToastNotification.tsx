import React from "react";
import {
  DeviceEventEmitter,
  SafeAreaView,
  StyleSheet,
  View
} from "react-native";
import ReactNativeHapticFeedback, {
  HapticFeedbackTypes
} from "react-native-haptic-feedback";
import Animated, {
  Easing,
  SequencedTransition,
  SlideInUp,
  SlideOutUp
} from "react-native-reanimated";
import { IOIcons, Icon } from "../core/icons";
import { CTA } from "../core/typography/CTA";
import { IOColors } from "../core/variables/IOColors";
import { IOAlertRadius } from "../core/variables/IOShapes";
import { Dismissable } from "./Dismissable";

type ToastVariant = "neutral" | "error" | "info" | "success" | "warning";

type Toast = {
  message: string;
  variant?: ToastVariant;
  icon?: IOIcons;
};

type ToastColors = {
  background: IOColors;
  stroke: IOColors;
};

const toastVariantToColors: Record<ToastVariant, ToastColors> = {
  neutral: {
    background: "turquoise-150",
    stroke: "turquoise-850"
  },
  error: {
    background: "error-100",
    stroke: "error-850"
  },
  info: {
    background: "info-100",
    stroke: "info-850"
  },
  success: {
    background: "success-100",
    stroke: "success-850"
  },
  warning: {
    background: "warning-100",
    stroke: "warning-850"
  }
};

const ToastNotification = ({ message, variant = "neutral", icon }: Toast) => {
  const colors = toastVariantToColors[variant];

  return (
    <View
      style={[
        styles.toast,
        {
          backgroundColor: IOColors[colors.background],
          borderColor: IOColors[colors.stroke]
        }
      ]}
      accessible={true}
      accessibilityRole={"alert"}
      accessibilityLabel={message}
    >
      <CTA color={colors.stroke} style={styles.content}>
        {message}
      </CTA>
      {icon && <Icon name={icon} size={24} color={colors.stroke} />}
    </View>
  );
};

/**
 * The event used to show a toast notification
 */
const TOAST_EVENT = "SHOW_TOAST";

/**
 * The maximum number of toasts that can be displayed at the same time
 * If the number of the toasts exceeds this number, the oldest one will be removed
 */
const MAX_TOAST_STACK_SIZE = 3;

/**
 * The time in milliseconds that a toast notification will be displayed
 */
const TOAST_DURATION_TIME = 5000;

/**
 * Toast type with an ID to be used as key in the list
 */
type ToastNotificationStackItem = Toast & { id: number };

type ToastNotificationStackItemProps = ToastNotificationStackItem & {
  onDismiss: () => void;
};

/**
 * A toast notification item that can be swiped to the right to dismiss it, with enter and exit animations
 */
const ToastNotificationStackItem = ({
  onDismiss,
  ...props
}: ToastNotificationStackItemProps) => (
  <Animated.View
    entering={SlideInUp.duration(300).easing(Easing.inOut(Easing.exp))}
    exiting={SlideOutUp.duration(300).easing(Easing.inOut(Easing.exp))}
    layout={SequencedTransition.duration(300)}
    style={{ paddingBottom: 8 }}
  >
    <Dismissable onDismiss={onDismiss}>
      <ToastNotification {...props} />
    </Dismissable>
  </Animated.View>
);

/**
 * Toast events payload
 */
type ToastEvent = Toast & {
  hapticFeedback?: keyof typeof HapticFeedbackTypes;
};

/**
 * A container that will display the toast notifications received by the {@link TOAST_EVENT} event
 */
const ToastNotificationContainer = () => {
  const [toasts, setToasts] = React.useState<
    ReadonlyArray<ToastNotificationStackItem>
  >([]);

  const handleToastEvent = (toast: ToastEvent) => {
    const id = new Date().getTime();

    setToasts(prevToasts => [{ id, ...toast }, ...prevToasts]);
    setTimeout(() => {
      setToasts(prevToasts => prevToasts.filter(t => t.id !== id));
    }, TOAST_DURATION_TIME);

    if (toast.hapticFeedback) {
      ReactNativeHapticFeedback.trigger(toast.hapticFeedback);
    }
  };

  const removeToastById = (id: number) => {
    setToasts(prevToasts => prevToasts.filter(t => t.id !== id));
  };

  const removeToastAtIndex = (index: number) => {
    setToasts(prevToasts => [
      ...prevToasts.slice(0, index),
      ...prevToasts.slice(index + 1)
    ]);
  };

  // If stack size exceed the maximum, remove the oldest toast
  React.useEffect(() => {
    if (toasts.length > MAX_TOAST_STACK_SIZE) {
      removeToastAtIndex(MAX_TOAST_STACK_SIZE);
    }
  }, [toasts]);

  /**
   * Subscribe to the toast event and add the new toast to the list
   */
  React.useEffect(() => {
    const subscription = DeviceEventEmitter.addListener(
      TOAST_EVENT,
      handleToastEvent
    );
    return () => {
      subscription.remove();
    };
  }, []);

  return (
    <SafeAreaView style={styles.container} pointerEvents="box-none">
      <View style={styles.list} pointerEvents="box-none">
        {toasts.map(toast => (
          <ToastNotificationStackItem
            key={toast.id}
            {...toast}
            onDismiss={() => removeToastById(toast.id)}
          />
        ))}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    bottom: 0,
    left: 0,
    right: 0,
    top: 0,
    overflow: "visible"
  },
  list: {
    padding: 24
  },
  toast: {
    borderRadius: IOAlertRadius,
    borderWidth: 1,
    padding: 16,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between"
  },
  content: {
    paddingVertical: 2
  }
});

/**
 * Options to customize the toast notification
 */
type IOToastOptions = Omit<ToastEvent, "message">;

/**
 * A toast notification service that can be used to show toast notifications
 */
const IOToast = {
  /**
   * Show a toast notification with the given message, variant and icon
   * @param message The message to display
   * @param variant A {@link ToastVariant} of the toast that will determine its colors, defaults to "neutral"
   * @param icon An optional {@link IOIcons} to display on the right of the toast
   *
   * @example
   *
   * IOToast.show("This is a toast notification", "info", "infoFilled");
   */
  show: (message: string, options?: IOToastOptions) => {
    DeviceEventEmitter.emit(TOAST_EVENT, {
      message,
      ...options
    });
  },
  /**
   * Show an error toast notification with the given message and an error icon
   * @param message The message to display
   *
   * @example
   *
   * IOToast.error("This is an error toast notification");
   */
  error: (message: string) => {
    IOToast.show(message, {
      variant: "error",
      icon: "errorFilled",
      hapticFeedback: "notificationError"
    });
  },
  /**
   * Show an info toast notification with the given message and an info icon
   * @param message The message to display
   *
   * @example
   *
   * IOToast.info("This is an info toast notification");
   */
  info: (message: string) => {
    IOToast.show(message, {
      variant: "info",
      icon: "infoFilled",
      hapticFeedback: "impactMedium"
    });
  },
  /**
   * Show a success toast notification with the given message and a check tick icon
   * @param message The message to display
   *
   * @example
   *
   * IOToast.success("This is a success toast notification");
   */
  success: (message: string) => {
    IOToast.show(message, {
      variant: "success",
      icon: "success",
      hapticFeedback: "notificationSuccess"
    });
  },
  /**
   * Show a warning toast notification with the given message and a warning sign icon
   * @param message The message to display
   *
   * @example
   *
   * IOToast.warning("This is a warning toast notification");
   */
  warning: (message: string) => {
    IOToast.show(message, {
      variant: "warning",
      icon: "warningFilled",
      hapticFeedback: "notificationWarning"
    });
  }
};

export { IOToast, ToastNotification, ToastNotificationContainer };
