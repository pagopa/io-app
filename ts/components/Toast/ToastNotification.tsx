import React from "react";
import { StyleSheet, View } from "react-native";
import { IOColors, Icon, IOAlertRadius } from "@pagopa/io-app-design-system";
import { CTA } from "../core/typography/CTA";
import { Toast, ToastVariant } from "./types";

type ColorVariant = {
  background: IOColors;
  stroke: IOColors;
};

const toastColorVariants: Record<ToastVariant, ColorVariant> = {
  neutral: {
    background: "turquoise-150",
    stroke: "turquoise-850"
  },
  error: {
    background: "error-100",
    stroke: "error-850"
  },
  info: {
    background: "info-100",
    stroke: "info-850"
  },
  success: {
    background: "success-100",
    stroke: "success-850"
  },
  warning: {
    background: "warning-100",
    stroke: "warning-850"
  }
};

type Props = Pick<Toast, "message" | "variant" | "icon">;

const ToastNotification = ({ message, variant = "neutral", icon }: Props) => {
  const colors = toastColorVariants[variant];

  return (
    <View
      style={[
        styles.toast,
        {
          backgroundColor: IOColors[colors.background],
          borderColor: IOColors[colors.stroke]
        }
      ]}
      accessible={true}
      accessibilityRole={"alert"}
      accessibilityLabel={message}
    >
      <CTA color={colors.stroke} style={styles.content}>
        {message}
      </CTA>
      {icon && <Icon name={icon} size={24} color={colors.stroke} />}
    </View>
  );
};

const styles = StyleSheet.create({
  toast: {
    borderRadius: IOAlertRadius,
    borderWidth: 1,
    padding: 16,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between"
  },
  content: {
    paddingVertical: 2
  }
});

export { ToastNotification };
