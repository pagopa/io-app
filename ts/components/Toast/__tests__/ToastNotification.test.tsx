import { render } from "@testing-library/react-native";
import React from "react";
import { ToastNotification } from "../ToastNotification";
import { Toast } from "../types";

describe("Test ToastNotification component", () => {
  it.each<Toast>([
    { message: "Hello", icon: "checkTick" },
    { message: "Hello", variant: "error" },
    { message: "Hello", variant: "info" },
    { message: "Hello", variant: "neutral" },
    { message: "Hello", variant: "success" },
    { message: "Hello", variant: "warning" }
  ])("should match snapshot for props (%s)", toast => {
    const component = render(<ToastNotification {...toast} />);
    expect(component).toMatchSnapshot();
  });
});
