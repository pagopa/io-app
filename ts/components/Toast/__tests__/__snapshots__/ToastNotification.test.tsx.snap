// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ToastNotification component should match snapshot for props ({ message: 'Hello', icon: 'checkTick' }) 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityRole="alert"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      },
      Object {
        "backgroundColor": "#AAEEEF",
        "borderColor": "#003B3D",
      },
    ]
  }
>
  <Text
    color="turquoise-850"
    defaultColor="black"
    defaultWeight="Regular"
    font="ReadexPro"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 20,
      }
    }
    style={
      Array [
        Object {
          "paddingVertical": 2,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20,
        },
        Object {
          "color": "#003B3D",
          "fontFamily": "Readex Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Hello
  </Text>
  <RNSVGSvgView
    accessibilityElementsHidden={true}
    accessibilityLabel=""
    accessible={false}
    align="xMidYMid"
    bbHeight={24}
    bbWidth={24}
    color={4278205245}
    focusable={false}
    height={24}
    importantForAccessibility="no-hide-descendants"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "color": "#003B3D",
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    tintColor={4278205245}
    vbHeight={24}
    vbWidth={24}
    width={24}
  >
    <RNSVGGroup>
      <RNSVGPath
        clipRule={0}
        d="M17.7071 9.29289C18.0976 9.68342 18.0976 10.3166 17.7071 10.7071L11.4142 17C10.6332 17.781 9.36684 17.7811 8.58579 17L5.29289 13.7071C4.90237 13.3166 4.90237 12.6834 5.29289 12.2929C5.68342 11.9024 6.31658 11.9024 6.70711 12.2929L10 15.5858L16.2929 9.29289C16.6834 8.90237 17.3166 8.90237 17.7071 9.29289Z"
        fill={
          Array [
            2,
          ]
        }
        fillRule={0}
        propList={
          Array [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`Test ToastNotification component should match snapshot for props ({ message: 'Hello', variant: 'error' }) 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityRole="alert"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      },
      Object {
        "backgroundColor": "#FFE0E0",
        "borderColor": "#761F1F",
      },
    ]
  }
>
  <Text
    color="error-850"
    defaultColor="black"
    defaultWeight="Regular"
    font="ReadexPro"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 20,
      }
    }
    style={
      Array [
        Object {
          "paddingVertical": 2,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20,
        },
        Object {
          "color": "#761F1F",
          "fontFamily": "Readex Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Hello
  </Text>
</View>
`;

exports[`Test ToastNotification component should match snapshot for props ({ message: 'Hello', variant: 'info' }) 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityRole="alert"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      },
      Object {
        "backgroundColor": "#E1F5FE",
        "borderColor": "#215C76",
      },
    ]
  }
>
  <Text
    color="info-850"
    defaultColor="black"
    defaultWeight="Regular"
    font="ReadexPro"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 20,
      }
    }
    style={
      Array [
        Object {
          "paddingVertical": 2,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20,
        },
        Object {
          "color": "#215C76",
          "fontFamily": "Readex Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Hello
  </Text>
</View>
`;

exports[`Test ToastNotification component should match snapshot for props ({ message: 'Hello', variant: 'neutral' }) 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityRole="alert"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      },
      Object {
        "backgroundColor": "#AAEEEF",
        "borderColor": "#003B3D",
      },
    ]
  }
>
  <Text
    color="turquoise-850"
    defaultColor="black"
    defaultWeight="Regular"
    font="ReadexPro"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 20,
      }
    }
    style={
      Array [
        Object {
          "paddingVertical": 2,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20,
        },
        Object {
          "color": "#003B3D",
          "fontFamily": "Readex Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Hello
  </Text>
</View>
`;

exports[`Test ToastNotification component should match snapshot for props ({ message: 'Hello', variant: 'success' }) 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityRole="alert"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      },
      Object {
        "backgroundColor": "#E1F4E1",
        "borderColor": "#224021",
      },
    ]
  }
>
  <Text
    color="success-850"
    defaultColor="black"
    defaultWeight="Regular"
    font="ReadexPro"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 20,
      }
    }
    style={
      Array [
        Object {
          "paddingVertical": 2,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20,
        },
        Object {
          "color": "#224021",
          "fontFamily": "Readex Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Hello
  </Text>
</View>
`;

exports[`Test ToastNotification component should match snapshot for props ({ message: 'Hello', variant: 'warning' }) 1`] = `
<View
  accessibilityLabel="Hello"
  accessibilityRole="alert"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      },
      Object {
        "backgroundColor": "#FFF5DA",
        "borderColor": "#614C15",
      },
    ]
  }
>
  <Text
    color="warning-850"
    defaultColor="black"
    defaultWeight="Regular"
    font="ReadexPro"
    fontStyle={
      Object {
        "fontSize": 16,
        "lineHeight": 20,
      }
    }
    style={
      Array [
        Object {
          "paddingVertical": 2,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 20,
        },
        Object {
          "color": "#614C15",
          "fontFamily": "Readex Pro",
          "fontStyle": "normal",
          "fontWeight": "400",
        },
      ]
    }
    weight="Regular"
  >
    Hello
  </Text>
</View>
`;
